content/projects/gsoc/2021/projects/git-credentials-binding.adoc

```diff
@@ -47,23 +49,34 @@ They will be able to create and delete remote branches.
 They will be able to use the git command line options of their choice, including `--single-branch` and `--recurse-submodules`.
 Users will be able to run authenticated git commands in their Jenkins Pipelines without modifying the git plugin.
 
-=== Implementation examples
+=== Implementation
+Both `gitUsernamePassword` and `gitSshPrivateKey` bindings use the https://plugins.jenkins.io/credentials/[Credential Plugin]
+to retrieve user's credential using the Credentials API, the https://plugins.jenkins.io/credentials-binding/[Credential Binnding Plugin]
+is used to bind Git specific environment variables with programs to perform authentication on behalf of the user, without their interaction with the command-line.
+
+==== Git Username and Password Binding
+
+The `gitUsernamePassword` implementation use the Jenkins username and password
+values retrieved through Credential API, to access a remote repository  over *HTTP* protocol. This binding use the
+`GIT_ASKPASS` environment variable to provide credentials requested by `Git CLI`, program specified by this variable is invoked with a suitable prompt on the command-line, and the user’s credential are read from its standard output.
+
+image::https://gist.githubusercontent.com/arpoch/96c75cc32afd0e50476e273bd0269a86/raw/7875f70561e1417721ec7f637fa6f160a27822a5/UsernameandPassword.png[Git Username and Password]
 
-The `sshPrivateKey` and `usernamePassword` implementations in the credentials binding plugin provide examples that create environment variables and temporary files for use by sh, bat, and powershell steps.
-Use them as a reference implementation.
-The git credentials implementation will need environment variables for the ssh and https credentials.
-The git credentials implementation will need files for the ssh credentials.
+IMPORTANT: By default, two variable bindings `GIT_USERNAME` and `GIT_PASSWORD`,
+bind with user's credential i.e. username and password respectively. These variable binding's are available in both freestyle project and pipeline script.
 
-The `gitSshPrivateKey` implementation should take an additional argument, `cliGitBaseVersion`, that is a string providing the minimum CLI git version that will be used in the sh, bat, and powershell commands within the `withCredentials` block.
+The `gitUsernamePassword` binding only provides authentication support for `Git CLI` implementaton and other Git implementations inside the `git client` plugin such as `JGit` and `JGit with Apache HTTP Client` are not supported. The variable bindings `GIT_USERNAME` and `GIT_PASSWORD` are accessible in `Git CLI`, JGit` and `JGit with Apache HTTP Client` Git implementations.
```

Comment:
Prefer two sentences.  Optional.  I removed the last sentence because I didn't understand what you were trying to describe there.  Can you describe that in more detail so that I better understand it?
```suggestion
The `gitUsernamePassword` binding only provides authentication support for the `Git CLI` implementation.
Other Git implementations inside the `git client` plugin such as `JGit` and `JGit with Apache HTTP Client` are not supported.
```